pkg/recommender/engine.go

```diff
@@ -206,6 +207,62 @@ func (e *Engine) RecommendClusterScaleOut(provider string, service string, regio
 	return e.RecommendCluster(provider, service, region, clReq, req.ActualLayout)
 }
 
+// RecommendClusters performs recommendation
+func (e *Engine) RecommendClusters(req Request) (map[string][]*ClusterRecommendationResp, error) {
+	respPerService := make(map[string][]*ClusterRecommendationResp)
```

Comment:
cahange/redesign the impelementation here as follows:
- first extract the parameters of the recomendation call into a slice of parameter struct (define it inline or as an unexported type) -use the muliple embedded loops only for this 
- iterate over this slice and call the recommendation / handle responses as required

Reply:
this is almost there, what i initially meant is something like this:
please note, that the limitation logic needs to be adjusted too! 
`
// RecommendMultiCluster performs recommendation
func (e *Engine) RecommendMultiCluster(req MultiClusterRecommendationReq) (map[string][]*ClusterRecommendationResp, error) {
	respPerService := make(map[string][]*ClusterRecommendationResp)

	type recParams struct {
		provider string
		service  string
		region   string
	}

	var rp []recParams = make([]recParams, 0, 0)

	// get the slice with all the parameters required for recommendation
	for _, provider := range req.Providers {
		for _, service := range provider.Services {
			regions, err := e.getRegions(provider.Provider, service, req)
			if err != nil {
				return nil, err
			}
			for _, region := range regions {
				// we gather parameters for all recommendations here
				rp = append(rp, recParams{provider: provider.Provider, service: service, region: region})
			}
		}
	}

	e.log.Debug("gathered parameters for multi cloud recommendations", map[string]interface{}{"calls": len(rp)})
	var responses []*ClusterRecommendationResp
	for _, rParams := range rp {
		if response, err := e.RecommendCluster(rParams.provider, rParams.service, rParams.region, req.ClusterRecommendationReq, nil); err != nil {
...................
`